src/plugins/discover/public/application/view_components/canvas/discover_chart_container.tsx

```diff
@@ -3,25 +3,84 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { i18n } from '@osd/i18n';
+
 import './discover_chart_container.scss';
-import React, { useMemo } from 'react';
+import React, { useMemo, useState } from 'react';
+import {
+  EuiButtonIcon,
+  EuiCompressedSwitch,
+  EuiContextMenu,
+  EuiPanel,
+  EuiPopover,
+} from '@elastic/eui';
 import { DiscoverViewServices } from '../../../build_services';
 import { useOpenSearchDashboards } from '../../../../../opensearch_dashboards_react/public';
 import { useDiscoverContext } from '../context';
 import { SearchData } from '../utils/use_search';
 import { DiscoverChart } from '../../components/chart/chart';
+import { QUERY_ENHANCEMENT_ENABLED_SETTING } from '../../../../common';
+import { getNewDiscoverSetting, setNewDiscoverSetting } from '../../components/utils/local_storage';
 
 export const DiscoverChartContainer = ({ hits, bucketInterval, chartData }: SearchData) => {
   const { services } = useOpenSearchDashboards<DiscoverViewServices>();
-  const { uiSettings, data, core } = services;
+  const { uiSettings, data, core, storage } = services;
   const { indexPattern, savedSearch } = useDiscoverContext();
+  const isEnhancementsEnabled = uiSettings.get(QUERY_ENHANCEMENT_ENABLED_SETTING);
```

Comment:
Nit: just for readability
```suggestion
  const isQueryEnhancementsEnabled = uiSettings.get(QUERY_ENHANCEMENT_ENABLED_SETTING);
```